### DISCUSSION TITLE
How do I remap a trailing slash?

### DISCUSSION THREAD

**Discussion by AndrewKvalheim** [lycheeverse/lychee#1375] [Q&A]
I’d like to perform this transformation, which I believe requires matching the trailing slash:

```
/     → /index.html
/a    → /a.html
/a/   → /a/index.html
/a/a  → /a/a.html
/a/a/ → /a/a/index.html
```

Surprisingly, something in lychee eats the trailing slash before I have a chance:

```console
$ echo '<a href='{/,/a,/a/,/a/a,/a/a/}'>' > a.html
$ lychee --dump --base . a.html
file:///tmp/demo
file:///tmp/demo/a
file:///tmp/demo/a/a
```

Five distinct paths went in, but only three came out. Is there some default remapping rule I need to disable first?

**Comment by mre** [comment]
I haven't looked into it, but I think it's because of incorrect path concatenation when providing a `base`. 
There are similar issues in the issue tracker. I think this part needs more tests to fix edge-cases. It is a somewhat ad-hoc implementation that was hand-rolled for lychee's needs, and it's not my best work, I'm afraid.

At least the `url` crate is not the culprit. It keeps the trailing slash, as can be seen in this [Rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2ee3abd349b986ddf241113e5a02d23f) example.

**Reply by mre** [reply]
Dropping a note to myself here to try https://github.com/danreeves/path-clean for path handling.
